lib/oop/computeShapes.dart

```diff
@@ -1,61 +1,183 @@
+          if (parameters == null || parameters.isEmpty) {
+            print('Parameters Height and Base are incorrect ');
+            continue;
+          }
+        }
+        var splitParameters = parameters.split(' ');
+        if (splitParameters.length < 2) {
+          print('Parameters Height and Base are incorrect ');
+          continue;
+        }
+        var heightString = splitParameters[0];
+        var baseString = splitParameters[1];
+        double? height = double.tryParse(heightString);
+        double? base = double.tryParse(baseString);
+        if (height == null || base == null) {
+          print('Parameters Height and Base are incorrect ');
+          continue;
+        }
+        var t = IsoscelesTriangle(name, height, base);
+        figures.add(t);
         break;
 
       case 'Rectangle':
-        print('Please enter parameters .... ');
-        // add shape name and input (parameters) to the list
+        String name = 'Rectangle';
+        String? parameters;
+        print('Please enter a and b sides of $name separated by a space');
+        parameters = stdin.readLineSync();
+        while (parameters == null || parameters.isEmpty) {
+          print(
+              'Please enter correct parameters a and b sides of $name separated by a space');
+          parameters = stdin.readLineSync();
+          if (parameters == null || parameters.isEmpty) {
+            print('Parameters a and b sides are incorrect');
+            continue;
+          }
+        }
+        var splitParameters = parameters.split(' ');
+        if (splitParameters.length < 2) {
+          print('Parameters a and b sides are incorrect ');
+          continue;
+        }
+        var aSideString = splitParameters[0];
+        var bSideString = splitParameters[1];
+        double? aSide = double.tryParse(aSideString);
+        double? bSide = double.tryParse(bSideString);
+        if (aSide == null || bSide == null) {
+          print('Parameters a and b sides are incorrect ');
+          continue;
+        }
+        var t = Rectangle(name, aSide, bSide);
+        figures.add(t);
         break;
 
       case 'Square':
-        print('Please enter parameters .... ');
-        // add shape name and input (parameters) to the list
+        String name = 'Square';
+        String? parameter;
+        print('Please enter Side of $name');
+        parameter = stdin.readLineSync();
+        while (parameter == null || parameter.isEmpty) {
+          print('Please enter correct parameter Side of $name');
+          parameter = stdin.readLineSync();
+          if (parameter == null || parameter.isEmpty) {
+            print('Parameter Side is incorrect ');
+            continue;
+          }
+        }
+        double? side = double.tryParse(parameter);
+        if (side == null) {
+          print('Parameter Side is incorrect ');
+          continue;
+        }
+        var t = Square(name, side);
+        figures.add(t);
         break;
 
```

Comment:
The previos inputs are so similar. Create a function that passing a label return the values of the parameters